catalog/app/containers/Admin/Table/Table.tsx

```diff
@@ -6,15 +6,56 @@ import { lighten } from '@material-ui/core/styles/colorManipulator'
 
 import useMemoEq from 'utils/useMemoEq'
 
-const changeDirection = (d) => (d === 'asc' ? 'desc' : 'asc')
+interface UseFilteringProps<Row> {
+  rows: readonly Row[]
+  filterBy: (r: Row) => string
+}
+
+export function useFiltering<Row>({ rows, filterBy }: UseFilteringProps<Row>) {
+  const [value, onChange] = React.useState('')
+  const filtered = React.useMemo(
+    () =>
+      value
+        ? rows.filter((row) => filterBy(row).toLowerCase().includes(value.toLowerCase()))
+        : rows,
+    [filterBy, value, rows],
+  )
+  return { value, onChange, filtered }
+}
 
-export function useOrdering({ rows, ...opts }) {
+type Direction = 'asc' | 'desc'
+
+const changeDirection = (d: Direction) => (d === 'asc' ? 'desc' : 'asc')
+
+export type Column<Row> = {
+  align?: M.TableCellProps['align']
+  getDisplay: $TSFixMe
+  getValue: $TSFixMe
```

Comment:
these two can be made at least a little bit stricter by typing them as `(r: Row) => $TSFixMe`

Reply:
I added these types:
```ts
+  getDisplay?: (v: $TSFixMe, r: Row, opts?: $TSFixMe) => React.ReactNode
+  getValue: (r: Row) => $TSFixMe
```

Reply:
Also, changed `const columns = [...] as Table.Column[]` to `const columns: Table.Column[] = [...]`